src/transforms/projectionpursuit.jl

```diff
@@ -0,0 +1,199 @@
+other coordinates in the rotated orthonormal basis `Q = [α ...]` are left untouched.
+
+The non-singularity of Q is controlled by assuring that norm(det(Q)) ≥ `tol`. The iterative 
+process terminates whenever the transformed samples are "more Gaussian" than `perc`% of `n`
+randomly generated samples from the standard multivariate Gaussian distribution, or when the 
+number of iterations reaches a maximum `maxiter`.
+
+# Examples
+
+```julia
+ProjectionPursuit()
+ProjectionPursuit(deg=10)
+ProjectionPursuit(perc=.85, n=50)
+ProjectionPursuit(tol=1e-4, maxiter=250, deg=5, perc=.95, n=100)
+```
+
+"""
+
+struct ProjectionPursuit{T} <: StatelessTableTransform
+  tol::T
+  maxiter::Int
+  deg::Int
+  perc::T
+  n::Int
+end
+
+ProjectionPursuit(;tol=1e-6, maxiter=100, deg=5, perc=.95, n=100) =
+  ProjectionPursuit{typeof(tol)}(tol, maxiter, deg, perc, n)
+
+isrevertible(::Type{<:ProjectionPursuit}) = true
+
+# transforms a row of random variables into a convex combination 
+# of random variables with values in [-1,1] and standard normal distribution
+function rscore(Z, α)
+  ᾱ = (1/norm(α)) .* α
+  X = Z * ᾱ
+  2 .* cdf.(Normal(), X) .- 1
+end
+
+# projection index of sample along a given direction
+function pindex(transform, Z, α)
+  d = transform.deg
+  X = rscore(Z, α)
+  I = (3/2) * mean(X)^2
+  if d > 1
+    Pⱼ₋₂, Pⱼ₋₁ = ones(size(X)), X
+    for j = 2:d
+      Pⱼ₋₂, Pⱼ₋₁ = 
+        Pⱼ₋₁, (1/j) * ((2j-1) * X .* Pⱼ₋₁ - (j-1) * Pⱼ₋₂)
+      I += ((2j+1)/2) * (mean(Pⱼ₋₁))^2
+    end
+  end
+  I
+end
+
+# projection index of the standard multivariate Gaussian
+function gindex(transform, N, q)
+  n  = transform.n
+  E  = 1.0 * I(q)
+  Iₙ = Matrix{Float64}(undef, n, q)
+  for i = 1:n
+    S = randn(N, q)
+    for j = 1:q
+      Iₙ[i,j] = pindex(transform, S, E[:,j])
+    end
+  end
+  Iₙ
+end
+
+function alphaguess(transform, Z)
+  q = size(Z, 2)
+  E = 1.0 * I(q)
+  
+  # objective function
+  func(α) = pindex(transform, Z, α)
+  
+  # evaluate score along axes
+  i  = argmax(func, eachcol(E))
+  α  = E[:,i]
+  Iₐ = func(α)
+  
+  # evaluate score along diagonals
+  diag(α, s, e) = (1/√(2+2s*α'*e)) * (α + s*e)
+  for eᵢ in eachcol(E)
+    d₊ = diag(α, +1, eᵢ)
+    d₋ = diag(α, -1, eᵢ)
+    f₊ = func(d₊)
+    f₋ = α'*eᵢ != 1.0 ? func(d₋) : 0.0
+    f, d = f₊ > f₋ ? (f₊, d₊) : (f₋, d₋)
+    if f > Iₐ
+      α  = d
+      Iₐ = f
+    end
+  end
+  
+  α
+end
+
+function neldermead(transform, Z, α₀)
+  f(α) = -pindex(transform, Z, α)
+  op = optimize(f, α₀)
+  minimizer(op)
+end
+
+function alphamax(transform, Z)
+  α = alphaguess(transform, Z)
+  neldermead(transform, Z, α)  
+end
+
+function rmstructure(transform, Z, α)
+  q = length(α)
+  Q, R = qr(hcat(α, rand(q, q-1)))
+  while norm(diag(R)) < transform.tol
+    Q, R = qr(hcat(α, rand(q, q-1)))
+  end
+  T = Z * Q
+  
+  # remove structure of first rotated axis
+  newtable, colcache = apply(Quantile(), Tables.table(T))
```

Comment:
```suggestion
  newtable, colcache = apply(Quantile(1), Tables.table(T))
```
Here you can apply the quantile for the first column only, right? Can you please double check that the code still works after this change?